File: 3765.complete-prime-number.py
#
# @lc app=leetcode id=3765 lang=python3
#
# [3765] Complete Prime Number
#
# @lc code=start
class Solution:
    def completePrime(self, num: int) -> bool:
        def is_prime(n):
            if n < 2:
                return False
            if n == 2:
                return True
            if n % 2 == 0:
                return False
            for i in range(3, int(n**0.5) + 1, 2):
                if n % i == 0:
                    return False
            return True
        
        # Convert to string to get prefixes and suffixes
        num_str = str(num)
        
        # Check all prefixes
        for i in range(1, len(num_str) + 1):
            prefix = int(num_str[:i])
            if not is_prime(prefix):
                return False
        
        # Check all suffixes
        for i in range(len(num_str)):
            suffix = int(num_str[i:])
            if not is_prime(suffix):
                return False
        
        return True
# @lc code=end

File: 3765.complete-prime-number.cpp
//
// @lc app=leetcode id=3765 lang=cpp
//
// [3765] Complete Prime Number
//

// @lc code=start
class Solution {
public:
    bool isPrime(int n) {
        if (n < 2) return false;
        if (n == 2) return true;
        if (n % 2 == 0) return false;
        for (int i = 3; (long long)i * i <= n; i += 2) {
            if (n % i == 0) return false;
        }
        return true;
    }
    
    bool completePrime(int num) {
        string s = to_string(num);
        int len = s.length();
        
        // Check all prefixes
        for (int i = 1; i <= len; i++) {
            int prefix = stoi(s.substr(0, i));
            if (!isPrime(prefix)) return false;
        }
        
        // Check all suffixes
        for (int i = 1; i <= len; i++) {
            int suffix = stoi(s.substr(len - i, i));
            if (!isPrime(suffix)) return false;
        }
        
        return true;
    }
};
// @lc code=end